src/VideoRemise/VideoRemise/FaveroFA01Trigger.cs

```diff
@@ -10,73 +10,132 @@ namespace VideoRemise
+    // The checksum byte is a simple unsigned sum of all of the initial data bytes. This means
+    // That it is possible for the checksum byte to be 0xff and look like a start of frame, so 
+    // the code needs to check for that. None of the other data bytes can be 0xff, so if teo
+    // consecutive flag values are seen, the first must be the checksum from the previous frame
+    // and the second the start of the new frame.
     internal class FaveroFA01Trigger : SerialTrigger
     {
         const byte StartOfFrame = 0xff;
-        const int LightStatusPosition = 5;
+        const int LightStatusPosition = 6;
         const byte LightStatusMask = 0x0f;
+        const int FrameSize = 10;
+        byte[] CurrentFrame;
 
         int position;
         Lights lightStatus;
 
         public FaveroFA01Trigger()
         {
-            readFrameLength = 10;
+            readFrameLength = FrameSize;
             position = 0;
             FiresClockEvents = false;
+            CurrentFrame = new byte[FrameSize];
         }
 
         public override async Task Initialize(VideoRemiseConfig config)
         {
             await base.Initialize(config);
-            device.BaudRate = 1200;
+            //device.BaudRate = 1200;
         }
 
-        protected override void ProcessBuffer()
+        protected override async Task ProcessBufferAsync()
         {
-            bool ScanForFrame()
+            /*async Task<bool>*/bool ScanForStartOfFrame()
             {
                 while (reader.UnconsumedBufferLength > 0)
                 {
                     var readByte = reader.ReadByte();
+                    //await Log($"Read {readByte:x} while looking for start of frame");
                     if (readByte == StartOfFrame)
                     {
+                        CurrentFrame[position] = readByte;
                         position++;
                         return reader.UnconsumedBufferLength > 0;
                     }
                 }
                 return false;
             }
 
+            bool ValidateChecksum()
+            {
+                byte sum = 0;
+                for (int i = 0; i < FrameSize - 1; i++)
+                {
+                    sum += CurrentFrame[i];
+                }
+                return sum == CurrentFrame[CurrentFrame.Length - 1];
+            }
+
             // If we think we're at the beginning, scan for a start of frame byte
             if (position == 0)
             {
-                if (!ScanForFrame())
+                if (!/*await*/ ScanForStartOfFrame())
                 {
-                    // If we didn't get passed the start-of-frame byte yet, keep reading
+                    // If we didn't get past the start-of-frame byte yet, keep reading
                     return;
                 }
             }
 
             var currentByte = reader.ReadByte();
-            position++;
+            //await Log($"Position {position}: read {currentByte:x}");
```

Comment:
Is this commented out because it is unhelpful for debugging, or because it was but no longer is? If it's the second, we may want to make a "debug mode" with additional logging.

Reply:
It got replaced by the easier to read debug logging down at about line 120. I've deleted this and some other old commented-out logging.